src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1232,6 +1235,61 @@ def forward(
             encoder_attentions=outputs.encoder_attentions,
         )
 
+    def generate(
+        self,
+        inputs: Optional[torch.Tensor] = None,
+        generation_config=None,
+        logits_processor=None,
+        stopping_criteria=None,
+        prefix_allowed_tokens_fn=None,
+        synced_gpus=False,
+        return_timestamps=None,
+        task=None,
+        is_multilingual=None,
+        **kwargs
+    ):
+        # priority: `generation_config` argument > `model.generation_config` (the default generation config)
+        if generation_config is None:
+            # legacy: users may modify the model configuration to control generation -- update the generation config
+            # model attribute accordingly, if it was created from the model config
+            if self.generation_config._from_model_config:
+                new_generation_config = GenerationConfig.from_model_config(self.config)
+                if new_generation_config != self.generation_config:
+                    warnings.warn(
+                        "You have modified the pretrained model configuration to control generation. This is a"
+                        " deprecated strategy to control generation and will be removed soon, in a future version."
+                        " Please use a generation configuration file (see"
+                        " https://huggingface.co/docs/transformers/main_classes/text_generation)I don't agree with"
+                        " this warning, the generation config can be different but the rest of the model is the"
+                        " samne.........."
+                    )
+                    self.generation_config = new_generation_config
+            generation_config = self.generation_config
```

Comment:
This part is redundant with `super.generate()`. Would be good if `self.generation_config` was already created and would only require an update at this point.